packages/back-end/src/controllers/experiments.ts

```diff
@@ -1521,17 +1523,41 @@ export async function postSnapshot(
   req.checkPermissions("runQueries", "");
 
   const { org } = getOrgFromReq(req);
+  const orgSettings = org.settings || {};
 
   let { statsEngine, regressionAdjustmentEnabled } = req.body;
   const { metricRegressionAdjustmentStatuses } = req.body;
   const { id } = req.params;
   const { phase, dimension } = req.body;
   const experiment = await getExperimentById(org.id, id);
 
-  statsEngine = (typeof statsEngine === "string" &&
-  ["bayesian", "frequentist"].includes(statsEngine)
+  if (!experiment) {
+    res.status(404).json({
+      status: 404,
+      message: "Experiment not found",
+    });
+    return;
+  }
+
+  if (!experiment.phases[phase]) {
+    res.status(404).json({
+      status: 404,
+      message: "Phase not found",
+    });
+    return;
+  }
+
+  let project = null;
+  if (experiment.project) {
+    project = await findProjectById(experiment.project, org.id);
+  }
+  const { settings: scopedSettings } = getScopedSettings(orgSettings, {
+    project: project?.settings,
+  });
+
+  statsEngine = ["bayesian", "frequentist"].includes(statsEngine + "")
     ? statsEngine
-    : org.settings?.statsEngine ?? "bayesian") as StatsEngine;
+    : scopedSettings.statsEngine.value;
```

Comment:
I'm not sure how else you'd do this logic:
- if a statsEngine is directly provided, use that instead of any internal setting
- otherwise fallback to the calculated setting

If your concern is that we are providing too many settings in the postSnapshot endpoint, perhaps, but that's definitely not something I want to alter right now. We're doing a snapshot model refactor in the near future so much of this will change.

Reply:
It's not a directed change I have in mind but just an observation that there are snapshot settings that this endpoint is accepting that perhaps we also need to include in `getScopedSettings`

Reply:
As in, we may need to add `snapshot` to our `ScopeDefinition`